### PR TITLE
Add clear mission support

### PR DESCRIPTION
**Pull request by douglaswsilva**
Implementation of [Mission: provide API to clear mission](https://github.com/mavlink/MAVSDK/issues/663)

### CODE REVIEW

**Comment by douglaswsilva** [comment]
Need to add Unit Tests. Let me know if you spot anything else.

File: src/plugins/mission/include/plugins/mission/mission.h [hunk 1]
```diff
@@ -182,6 +182,13 @@ class Mission : public PluginBase {
      */
     void pause_mission_async(result_callback_t callback);
 
+    /**
+     * @brief Clears the mission from vehicle (asynchronous).
```
Comment (julianoes) [inline src/plugins/mission/include/plugins/mission/mission.h]:
```suggestion
     * @brief Clears the mission saved on the vehicle (asynchronous).
```

**Comment by julianoes** [comment]
@douglaswsilva check the Jenkins build issues though.

**Comment by julianoes** [comment]
@douglaswsilva it would be nice if you could add the clear command to the integration test, e.g. at the end of the mission test you could clear the mission:

https://github.com/mavlink/MAVSDK/blob/0280deb7347e4d40bf6e84b655fb9fddb09270fa/src/integration_tests/mission.cpp#L252

**Comment by julianoes** [comment]
I'm just trying to understand why the integration test fails:

```
[11:50:20|Info ] [0mCleared mission, exiting. (mission.cpp:261)
[34m[11:50:20[34m|Info ] [0mMission accepted (mission_impl.cpp:166)
[34m[11:50:20|Info ] [0mDownloading mission... (mission.cpp:183)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[11:50:20|Info ] [0mMission uploaded. (mission.cpp:173)
[32m[11:50:20|Debug] [0mMAVLink: critical: IGN REQUEST LIST: Busy (system_impl.cpp:306)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mRetrying requesting mission list... (mission_impl.cpp:1401)
[32m[11:50:20|Debug] [0mMAVLink: critical: IGN REQUEST LIST: Busy (system_impl.cpp:306)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mRetrying requesting mission list... (mission_impl.cpp:1401)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:20|Warn ] [0mIgnoring mission request int, not active (mission_impl.cpp:109)
[33m[11:50:21|Warn ] [0mRetrying requesting mission list... (mission_impl.cpp:1401)
[32m[11:50:21|Debug] [0mRequested mission item 0 (mission_impl.cpp:1007)
[31m[11:50:21|Error] [0mError: unknown mission ack: 1 (mission_impl.cpp:178)
[33m[11:50:21|Warn ] [0mCallback is not set (mission_impl.cpp:1216)
[32m[11:50:21|Debug] [0mMAVLink: critical: IGN REQUEST LIST: Busy (system_impl.cpp:306)
[32m[11:50:21|Debug] [0mMAVLink: critical: Operation timeout (system_impl.cpp:306)
[32m[11:50:21|Debug] [0mMAVLink: critical: WPM: REJ. CMD: Req. WP was unexpected (system_impl.cpp:306)
../../../../src/integration_tests/mission.cpp:206: Failure
Expected equality of these values:
  status
    Which is: 4-byte object <01-00 00-00>
  std::future_status::ready
    Which is: 4-byte object <00-00 00-00>
[  FAILED  ] SitlTest.MissionAddWaypointsAndFly (38327 ms)
```

File: src/plugins/mission/mission_impl.cpp [hunk 2]
```diff
@@ -1063,6 +1063,61 @@ void MissionImpl::pause_mission_async(const Mission::result_callback_t& callback
...
+    mavlink_message_t message;
+    mavlink_msg_mission_clear_all_pack(
+        _parent->get_own_system_id(),
+        _parent->get_own_component_id(),
+        &message,
+        _parent->get_system_id(),
+        _parent->get_autopilot_id(),
+        MAV_MISSION_TYPE_MISSION);
+
+    _parent->send_message(message);
```
Comment (julianoes) [inline src/plugins/mission/mission_impl.cpp]: Found the mistake! Remove this line, otherwise you send the message twice!

Comment (JonasVautherin) [reply src/plugins/mission/mission_impl.cpp]: Good catch!

**Comment by douglaswsilva** [comment]
@julianoes Let me know if there's anything else I need to do for the PR.

**Pull request merged**